controllers/user.go

```diff
@@ -375,6 +376,20 @@ func (c *ApiController) AddUser() {
 		return
 	}
 
+	if user.Phone != "" {
```

Comment:
imho, same code should be apply in UpdateUser controller maybe you should pack these code as  a function in check.go name `CheckUserEmailAndPhone`

Reply:
object/check.go:532

These checks are already implemented in UpdateUser, however first level `if` is not same, so i am confused how to re-use this code in any other way.

Reply:
emmmm, maybe we can create a empty user object and pass it to function `CheckUpdateUser`

Reply:
Example
```go
func (c *ApiController) AddUser() {
	var user object.User
	err := json.Unmarshal(c.Ctx.Input.RequestBody, &user)
	if err != nil {
		c.ResponseError(err.Error())
		return
	}

	count, err := object.GetUserCount("", "", "", "")
	if err != nil {
		c.ResponseError(err.Error())
		return
	}

	if err := checkQuotaForUser(int(count)); err != nil {
		c.ResponseError(err.Error())
		return
	}

	emptyUser := object.User{}
	if msg := object.CheckUpdateUser(&emptyUser, &user, c.GetAcceptLanguage()); msg != "" {
		c.ResponseError(msg)
		return
	}

	c.Data["json"] = wrapActionResponse(object.AddUser(&user))
	c.ServeJSON()
}
```

Reply:
Well, if you think this is OK solution